packages/compass-import-export/src/components/import-preview.tsx

```diff
@@ -200,6 +255,58 @@ function MixedWarning({
   );
 }
 
+function TypeWarning({
+  result,
+  selectedType,
+}: {
+  result: CSVField;
+  selectedType: CSVParsableFieldType;
+}) {
+  const example = findBrokenCSVTypeExample(result.types, selectedType);
+
+  if (!example) {
+    return null;
+  }
+
+  const value =
+    example.firstValue.length < MAX_STRING_LENGTH
+      ? example.firstValue
+      : `${example.firstValue.slice(0, MAX_STRING_LENGTH)}…`;
+
+  return (
+    <Tooltip
+      align="top"
+      justify="middle"
+      delay={500}
+      trigger={({ children, ...props }) => (
+        <div {...props}>
+          {children}
+          <WarningIcon />
+        </div>
+      )}
+    >
+      <>
+        <Body as="p">This field has these detected types:</Body>
```

Comment:
Isn't `Body` by default a `p` element? We might not need we `as` property here (and below).

Reply:
I removed it now. I had some issue where it added extra css you can't remove at some point, but can't recreate it now so was probably just confused.